src/main/java/com/es/phoneshop/model/dao/ArrayListProductDao.java

```diff
@@ -19,10 +22,13 @@ public ArrayListProductDao() {
     @Override
     public Product getProduct(Long id) throws ProductNotFoundException {
         synchronized (lock) {
-            return products.stream().
-                    filter(product -> id.equals(product.getId()))
+            if(id == null) {
+                throw new ProductNotFoundException("Given id is null when using getProduct");
+            }
+            return products.stream()
+                    .filter(product -> id.equals(product.getId()))
                     .findAny()
-                    .orElseThrow(ProductNotFoundException::new);
+                    .orElseThrow(() -> new ProductNotFoundException("No product with such id when using getProduct"));
```

Comment:
Put more info in exception message. For instance - **Couldn't find product with id: 1**
Check **String.format()** for creating messages with parameters.